hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/StructuredTraceGraphBuilder.java

```diff
@@ -37,10 +37,12 @@ public static StructuredTraceGraph buildGraph(StructuredTrace trace) {
 
     if (shouldRebuildTraceEventsGraph || shouldRebuildTraceEntitiesGraph) {
       Instant start = Instant.now();
-      StructuredTraceGraph graph =
-          shouldRebuildTraceEventsGraph
-              ? StructuredTraceGraph.reCreateTraceEventsGraph(trace)
-              : StructuredTraceGraph.reCreateTraceEntitiesGraph(trace);
+      StructuredTraceGraph graph = cachedGraphThreadLocal.get();
+      if (shouldRebuildTraceEventsGraph) {
+        graph.reCreateTraceEventsGraph(trace);
+      } else {
+        graph.reCreateTraceEntitiesGraph(trace);
+      }
       LOG.debug(
```

Comment:
Same as above.
wrap the below debug statement inside a LOG.isDebugEnabled()

Reply:
Done